cmds/wget/wget.go

```diff
@@ -53,8 +59,36 @@ func main() {
 		usage()
 	}
 
-	url := flag.Arg(0)
-	if err := wget(url, os.Stdout); err != nil {
+	url, err := url.Parse(flag.Arg(0))
+	if err != nil {
+		log.Fatalf("%v\n", err)
+	}
+
+	var out io.Writer
+
+	if len(*outPath) == 0 {
+		fileName := "index.html"
+		if url.Path[len(url.Path)-1] != '/' {
```

Comment:
Technically we can get an index out of bounds with this by doing wget "". I suggest checking the url length first after Parse and just log.Fatalf out of theres no URL

Reply:
Otherwise, everything else looks good to me!